### PR TITLE
More logs - uniform accross the helpers set

### PR DESCRIPTION
**Pull request by vladbalanescu**
Changes:
---
- [x] more unifrom logging accross the board for all of our helpers

Discsussion:
---
@PeterNgTr I think this PR is a valuable one since it'll give the user more chances to see what's wrong if anything happens in any helper, both error and success cases. You can see I have included stuff like `Success` and `Error` messages alongside helpers names to give our consumers a better insight into what's going wrong, in case that happens when they're trying to update their TR projects.

### CODE REVIEW

File: lib/testrail.js [hunk 1]
```diff
@@ -24,102 +23,131 @@ class TestRail {
 	async getCases(projectId, suiteId) {
 		try {
 			const res = await this.axios.get(`get_cases/${projectId}&suite_id=${suiteId}`);
+			output.log(`getCases: SUCCESS - the request data is ${JSON.stringify(data)}`);
```
Comment (PeterNgTr) [inline lib/testrail.js:26]: I guess we don't need this for GET request right?

Comment (vladbalanescu) [reply lib/testrail.js:26]: erm...maybe not, but I thought I might as well do it to keep it in line with the rest?

File: lib/testrail.js [hunk 2]
```diff
@@ -24,102 +23,131 @@ class TestRail {
...
 		try {
 			const res = await this.axios.get('get_suites/' + projectId);
+			output.log(`getSuites: SUCCESS - the request data is ${JSON.stringify(data)}`);
+			output.log(`getSuites: SUCCESS - the response data is ${JSON.stringify(res.data)}`);
 			return res.data;
 		} catch (error) {
-			output.error(`Cannot get suites due to ${error}`);
+			const parsedError = error && error.response && error.response.data ? error.response.data.error : error;
+			output.error(`getSuites: ERROR - cannot get results for case ${caseId} on run ${runId} due to ${parsedError}`);
+			output.error(`getSuites: ERROR - request data was ${JSON.stringify(data)}`);
```
Comment (PeterNgTr) [inline lib/testrail.js:71]: I guess we dont have data here :D

File: lib/testrail.js [hunk 3]
```diff
@@ -24,102 +23,131 @@ class TestRail {
...
 		}
 	}
 
 	async getConfigs(projectId) {
 		try {
 			const res = await this.axios.get('get_configs/' + projectId);
+			output.log(`getConfigs: SUCCESS - the request data is ${JSON.stringify(data)}`);
```
Comment (PeterNgTr) [inline lib/testrail.js:78]: I guess we dont have data here :D

File: lib/testrail.js [hunk 4]
```diff
@@ -24,102 +23,131 @@ class TestRail {
...
+			output.log(`getConfigs: SUCCESS - the response data is ${JSON.stringify(res.data)}`);
 			return res.data;
 		} catch (error) {
-			output.error(`Cannot get configs due to ${error}`);
+			const parsedError = error && error.response && error.response.data ? error.response.data.error : error;
+			output.error(`getConfigs: ERROR - cannot get results for case ${caseId} on run ${runId} due to ${parsedError}`);
+			output.error(`getConfigs: ERROR - request data was ${JSON.stringify(data)}`);
```
Comment (PeterNgTr) [inline lib/testrail.js:84]: I guess we dont have data here :D

File: lib/testrail.js [hunk 5]
```diff
@@ -24,102 +23,131 @@ class TestRail {
...
+			const parsedError = error && error.response && error.response.data ? error.response.data.error : error;
+			output.error(`addPlanEntry: ERROR - cannot get results for case ${caseId} on run ${runId} due to ${parsedError}`);
+			output.error(`addPlanEntry: ERROR - request data was ${JSON.stringify(data)}`);
 		}
 	}
 
 	async getSuites(projectId) {
 		try {
 			const res = await this.axios.get('get_suites/' + projectId);
+			output.log(`getSuites: SUCCESS - the request data is ${JSON.stringify(data)}`);
```
[the remark this answers is not in the document]
Comment (vladbalanescu) [reply lib/testrail.js:65]: good catch, I'll fix this one right away 👍

File: lib/testrail.js [hunk 6]
```diff
@@ -24,102 +23,131 @@ class TestRail {
...
+			output.error(`updateRun: ERROR - cannot get results for case ${caseId} on run ${runId} due to ${parsedError}`);
+			output.error(`updateRun: ERROR - request data was ${JSON.stringify(data)}`);
 		}
 	}
 
 	async getResultsForCase(runId, caseId) {
 		return this.axios.get(`get_results_for_case/${runId}/${caseId}`).then((res) => {
-			output.log(`The response is ${JSON.stringify(res.data)}`);
-			output.log(`The case ${caseId} on run ${runId} is updated`);
+			output.log(`getResultsForCase: SUCCESS - the request data is ${JSON.stringify(data)}`);
```
Comment (PeterNgTr) [inline lib/testrail.js:116]: I guess we dont have data here :D

File: lib/testrail.js (hunk 2 above)
Comment (vladbalanescu) [reply lib/testrail.js:71]: good eyes again 😓

File: lib/testrail.js [hunk 7]
```diff
@@ -24,102 +23,131 @@ class TestRail {
...
+			output.log(`getResultsForCase: SUCCESS - the response data is ${JSON.stringify(res.data)}`);
 			return res.data;
 		}).catch(error => {
-			output.error(`Cannot get results for case ${caseId} on run ${runId} due to ${error}`);
+			const parsedError = error && error.response && error.response.data ? error.response.data.error : error;
+			output.error(`getResultsForCase: ERROR - cannot get results for case ${caseId} on run ${runId} due to ${parsedError}`);
+			output.error(`getResultsForCase: ERROR - request data was ${JSON.stringify(data)}`);
```
Comment (PeterNgTr) [inline lib/testrail.js:122]: I guess we dont have data here :D

File: lib/testrail.js [hunk 8]
```diff
@@ -24,102 +23,131 @@ class TestRail {
...
+			output.error(`addResultsForCases: ERROR - request data was ${JSON.stringify(data)}`);
 		});
 	}
 
 	async addAttachmentToResult(resultId, imageFile) {
 		let form = new FormData();
 		form.append('attachment', fs.createReadStream(path.join(global.output_dir, imageFile.toString())));
 
-		await this.axios({
+		this.axios({
```
Comment (PeterNgTr) [inline lib/testrail.js:142]: I'm not sure we still need await here?

File: lib/testrail.js (hunk 7 above)
Comment (vladbalanescu) [reply lib/testrail.js:122]: yep, correct

File: lib/testrail.js (hunk 8 above)
Comment (vladbalanescu) [reply lib/testrail.js:142]: oups, not sure how that happened. Wrong copy/paste possibly

File: lib/testrail.js [hunk 9]
```diff
@@ -24,102 +23,131 @@ class TestRail {
...
 			method: 'post',
 			data: form,
 			url: 'add_attachment_to_result/' + resultId,
 			headers: form.getHeaders()
 		}).catch(err => {
-			output.error(`Cannot attach file due to ${err}`);
+			const parsedError = error && error.response && error.response.data ? error.response.data.error : error;
+			output.error(`addAttachmentToResult: ERROR - cannot add attachment due to ${parsedError}`);
+			output.error(`addAttachmentToResult: ERROR - request data was ${JSON.stringify(data)}`);
```
Comment (PeterNgTr) [inline lib/testrail.js:150]: I assume this shall be `form` here?

File: lib/testrail.js (hunk 1 above)
Comment (PeterNgTr) [reply lib/testrail.js:26]: my gut feeling that the data is here is null :D

**Comment by vladbalanescu** [comment]
@PeterNgTr All comments addressed, mind re-checking please? 🙃

File: lib/testrail.js (hunk 1 above)
Comment (vladbalanescu) [reply lib/testrail.js]: and what if I tell you that might be slightly wrong? 😓 
image

File: lib/testrail.js [hunk 10]
```diff
@@ -24,88 +23,107 @@ class TestRail {
 	async getCases(projectId, suiteId) {
 		try {
 			const res = await this.axios.get(`get_cases/${projectId}&suite_id=${suiteId}`);
+			output.log(`getCases: SUCCESS - the response data is ${JSON.stringify(res.data)}`);
 			return res.data;
 		} catch (error) {
 			const parsedError = error && error.response && error.response.data ? error.response.data.error : error;
-			output.error(`Cannot get cases for projectId:${projectId} & suiteId:${suiteId}, due to ${parsedError}`);
+			output.error(`getCases: ERROR - cannot get results for case ${caseId} on run ${runId} due to ${parsedError}`);
```
Comment (PeterNgTr) [inline lib/testrail.js:30]: maybe I'm wrong here, but is this the case that caseId or runId would be null here 😨

File: lib/testrail.js [hunk 11]
```diff
@@ -24,88 +23,107 @@ class TestRail {
...
 	async addPlan(projectId, data) {
 		try {
 			const res = await this.axios.post('add_plan/' + projectId, data);
+			output.log(`addPlan: SUCCESS - the request data is ${JSON.stringify(data)}`);
+			output.log(`addPlan: SUCCESS - the response data is ${JSON.stringify(res.data)}`);
 			return res.data;
 		} catch (error) {
-			output.error(`Cannot add new plan due to ${error}`);
+			const parsedError = error && error.response && error.response.data ? error.response.data.error : error;
+			output.error(`addPlan: ERROR - cannot get results runId:${runId} due to ${parsedError}`);
```
Comment (PeterNgTr) [inline lib/testrail.js:42]: same thought here.

File: lib/testrail.js (hunk 10 above)
Comment (vladbalanescu) [reply lib/testrail.js:30]: ah no, unpushed, commit, sorry!

**Comment by vladbalanescu** [comment]
@PeterNgTr Done again. Should be good this time 🙃

File: lib/testrail.js (hunk 1 above)
Comment (PeterNgTr) [reply lib/testrail.js]: :D I hardly see a GET call with payload

**Pull request merged**